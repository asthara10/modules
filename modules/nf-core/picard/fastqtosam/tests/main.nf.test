nextflow_process {

    name "Test Process PICARD_FASTQTOSAM"
    script "../main.nf"
    config "./nextflow.config"
    process "PICARD_FASTQTOSAM"

    tag "modules"
    tag "modules_nfcore"
    tag "picard"
    tag "picard/fastqtosam"

    test("test_picard_fastqtosam_single") {


        when {
            process {
                """
                
                input[0] = [
        [ id:'test', single_end:true ], // meta map
        [
            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
        ]
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_picard_fastqtosam_paired") {
        when {
            process {
                """
                        input[0] = [
        [ id:'test', single_end:false ], // meta map
        [
            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
        ]
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    test("test_picard_fastqtosam_paired_custom_samplename") {
        when {
            process {
                """
                        input[0] = [
        [ id:'test', single_end:false ], // meta map
        [
            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
        ]
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
