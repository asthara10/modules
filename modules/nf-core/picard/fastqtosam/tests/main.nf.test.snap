{
    "versions": {
        "content": [
            [
                "versions.yml:md5,5a6d1ee6d93cfe32fedd55e713a7d9e6"
            ]
        ],
        "timestamp": "2023-11-24T13:05:06.83185"
    }
}